### ISSUE TITLE
Configure egress for different external TCP service with same port

### ISSUE THREAD

**Issue by kinash000** [istio/istio#24937]
**Bug description**
When TCP service entry is set without an address field and the resolution is set to DNS then all connections on port defined in service entry not routed by host.
I also tried defining the service as ExternalService instead of ServiceEntry, but the result was similar
if i use tls then everything is fine.

**SE1:**
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: kafka-se
spec:
  exportTo:
    - .
  hosts:
    - kvm1-synapse-kafka.ru
  location: MESH_EXTERNAL
  ports:
    - name: tcp
      number: 9092
      protocol: TCP
  resolution: DNS
```
**SE2:**
```
apiVersion: networking.istio.io/v1alpha3
kind: ServiceEntry
metadata:
  name: kafka-se2
spec:
  exportTo:
    - .
  hosts:
    - kafka-x86-kf-new-3.ru
  location: MESH_EXTERNAL
  ports:
    - name: tcp
      number: 9092
      protocol: TCP
  resolution: DNS
```
We have two applications that establish external tcp connections to different hosts, but the same ports.
As a result, both applications invoke the same host (istio-proxy log):
[2020-06-24T09:13:38.253Z] "- - -" 0 - "-" 107318 31100 687421 - "-" "-" "-" "-" "10.116.16.82:9092" outbound|9092||kvm1-synapse-kafka.ru 10.128.49.66:59092 10.116.16.82:9092 10.128.49.66:59090 -

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

[ ] Multi Cluster
[X] Virtual Machine (Cluster raised on Virtual Machines)
[ ] Multi Control Plane

**Expected behavior**
Connect to a specific server by host:port

**Steps to reproduce the bug**
Create two TCP-ServiceEntry with same port for different hosts

**Version (include the output of `istioctl version --remote` and `kubectl version` and `helm version` if you used Helm)**
Istio 1.4.6 in OpenShift 4 (Kubernetes 1.16.2)

**Comment by kinash000** [comment]
past issue was closed: https://github.com/istio/istio/issues/8833

**Comment by howardjohn** [comment]
This is not possible in Envoy. TCP services do not have a Host (its an HTTP header). The only information we have is port and IP address. Since you have not provided an Address (https://istio.io/latest/docs/reference/config/networking/service-entry/#ServiceEntry) on the ServiceEntry, we cannot distinguish between the two

**Comment by savinov** [comment]
Actually, it is possible since this PR is merged into master: #24919  #26826 
That's available in istio 1.8 with preview profile or istio 1.9 by default.
